ConfigSpace/hyperparameters.pyx

```diff
@@ -258,6 +258,38 @@ cdef class Constant(Hyperparameter):
                       transform: bool = False) -> List:
         return []
 
+    def pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the original parameter space (the one specified by the user).
+        For each parameter type, there is also a method _pdf which 
+        operates on the transformed (and possibly normalized) parameter
+        space. Only legal values return a positive probability density,
+        otherwise zero.
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        if vector.ndim != 1:
+            raise ValueError("Method pdf expects a one-dimensional numpy array")
+        return self._pdf(vector)
+
+    def _pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the transformed (and possibly normalized, depends on the parameter 
+        type) space. As such, one never has to worry about log-normal 
+        distributions, only normal distributions (as the inverse_transform
+         in the pdf method handles these).
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        return (vector == self.value).astype(float)
+
+    def get_max_density(self):
```

Comment:
Hey, what's the usecase of this?

Reply:
Good question! When running PiBO with RandomForests and continous parameters, one needs to discretize the pdf to avoid a pathological case where BO just gets stuck evaluating one point over and over. The short explanation for this is that a RF must have a piecewise continous acquisition function to actually work as intended.

To accomplish the discretized prior, it gets binned to certain threshold between its minimum and maximum. This is simply done by
 `np.linspace(0, max_value, some_number of bins)`
, where
`max_value = param.get_max_density()`
Hopefully, that clarifies it somewhat.